ydb/mvp/oidc_proxy/oidc_protected_page_nebius.cpp

```diff
@@ -19,29 +18,42 @@ THandlerSessionServiceCheckNebius::THandlerSessionServiceCheckNebius(const NActo
     : THandlerSessionServiceCheck(sender, request, httpProxyId, settings)
 {}
 
-void THandlerSessionServiceCheckNebius::StartOidcProcess(const NActors::TActorContext& ctx) {
-    NHttp::THeaders headers(Request->Headers);
-    LOG_DEBUG_S(ctx, EService::MVP, "Start OIDC process");
-
-    NHttp::TCookies cookies(headers.Get("Cookie"));
-    TString sessionCookieName = CreateNameSessionCookie(Settings.ClientId);
-    TStringBuf sessionCookieValue = cookies.Get(sessionCookieName);
-    if (!sessionCookieValue.Empty()) {
-        LOG_DEBUG_S(ctx, EService::MVP, "Using session cookie (" << sessionCookieName << ": " << NKikimr::MaskTicket(sessionCookieValue) << ")");
+TStringBuf THandlerSessionServiceCheckNebius::GetCookie(const NHttp::TCookies& cookies, const TString& cookieName, const NActors::TActorContext& ctx) {
+    TStringBuf cookieValue = cookies.Get(cookieName);
+    if (!cookieValue.Empty()) {
+        LOG_DEBUG_S(ctx, EService::MVP, "Using cookie (" << cookieName << ": " << NKikimr::MaskTicket(cookieValue) << ")");
     }
+    return cookieValue;
+}
 
-
-    TString sessionToken;
+TString THandlerSessionServiceCheckNebius::DecodeToken(const TStringBuf& cookie, const NActors::TActorContext& ctx) {
+    TString token;
     try {
-        Base64StrictDecode(sessionCookieValue, sessionToken);
+        Base64StrictDecode(cookie, token);
     } catch (std::exception& e) {
-        LOG_DEBUG_S(ctx, EService::MVP, "Base64Decode session cookie: " << e.what());
-        sessionToken.clear();
+        LOG_DEBUG_S(ctx, EService::MVP, "Base64Decode " << cookie << " cookie: " << e.what());
+        token.clear();
     }
+    return token;
+}
```

Comment:
There is this code in `ydb/mvp/oidc_proxy/oidc_impersonate_start_page_nebius.h` May be extract it to common function?
THandlerSessionServiceCheckNebius::GetCookie the same